File: max-profit/max-profit.js
let maxProfit = function(prices) {
    let buy = prices[0]
    let profit = 0

    for (let i = 0; i < prices.length; i++) {
        let price = prices[i]
        if (price < buy) {
            buy = price
        } else if (price - buy > profit) {
            profit = price - buy
        }
    }

    return profit
};

File: max-profit/max-profit.go
package main

func maxProfit(prices []int) int {
	profit := 0
	buy := prices[0]

	for i := 1; i < len(prices); i++ {
		current := prices[i]

		if current < buy {
			buy = current
			continue
		}

		potentialProfit := current - buy
		if potentialProfit > profit {
			profit = potentialProfit
		}
	}

	return profit
}
